src/main/java/com/arangodb/velocypack/VPackStringSlice.java

```diff
@@ -0,0 +1,46 @@
+package com.arangodb.velocypack;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Wrapper around a {@link ValueType.STRING} supporting fast bytewise comparison.
+ *
+ * @see https://github.com/arangodb/velocypack/blob/master/VelocyPack.md#objects
+ */
+public class VPackStringSlice implements Comparable<VPackStringSlice> {
+    private byte[] vpack;
+    /**
+     * Index of the string bytes within {@link vpack},
+     * i.e. tag byte and length are somewhere before this index.
+     */
+    private int start;
```

Comment:
I wasn't sure about this decision. Starting at the tag byte would make it more similar to `VPackSlice`, but it would complicate the `compareToBytes` operations. Those comparisons are on the hot path for the `O(ln n)` path queries and for sorting the offset table while constructing the object.

I couldn't think of a good way to make this class private, so it's probably worth giving it some thought before merging.

@mpv1989 @hkernbach What do you think?